apps/studio/src/components/sales/PingSaleComplete.tsx

```diff
@@ -0,0 +1,126 @@
+import { FunctionComponent, useCallback, useEffect, useState } from "react";
+import { SaleStatus } from "@newm-web/types";
+import {
+  FIFTEEN_SECONDS_IN_MILLISECONDS,
+  FIVE_MINUTES_IN_MILLISECONDS,
+} from "@newm-web/utils";
+import { useGetSalesQuery } from "../../modules/sale";
+import {
+  LOCAL_STORAGE_SALE_COMPLETE_PENDING_KEY,
+  SALE_COMPLETE_UPDATED_EVENT,
+} from "../../common";
+
+const PingSaleComplete: FunctionComponent = () => {
+  const [currentPollingInterval, setPollingInterval] = useState<number>();
+  const [saleCompleteSongIds, setSaleCompleteSongIds] = useState<
+    string[] | null
+  >();
+  const [timeoutId, setTimeoutId] = useState<NodeJS.Timeout | null>(null);
+
+  const { data: sales = [], isLoading: isGetSalesLoading } = useGetSalesQuery(
+    {
+      saleStatuses: [SaleStatus.SoldOut],
+      songIds: saleCompleteSongIds || undefined,
+    },
+    {
+      pollingInterval: currentPollingInterval,
+      skip:
+        saleCompleteSongIds === null ||
+        (saleCompleteSongIds && saleCompleteSongIds.length === 0),
+    }
+  );
+
+  const handleSaleComplete = useCallback(() => {
+    const pendingSales = localStorage.getItem(
```

Comment:
Nothing to block here just curious why the use of `localStorage` vs our utils `LocalStorage`

https://github.com/projectNEWM/newm-web/blob/22658d7fcf36045253dcccdb1bbdbbc91c751f74/packages/utils/src/lib/storage.ts#L5

Reply:
Most of this code was taken from the [PingSaleEnd](https://github.com/projectNEWM/newm-web/blob/master/apps/studio/src/components/sales/PingSaleEnd.tsx), which used `localStorage`, so it was just from copying and pasting. I agree we should use the util in new code though.

Reply:
Going to leave it as is for now, just because I've already tested everything and it's a high priority fix. I can update it in a future PR.